### ISSUE TITLE
Reduce model size

### ISSUE THREAD

**Issue by eraser3112** [kendryte/nncase#80]
Convert my own model to .kmodel from .h5(based yolov2-tiny).
Use "toco" to convert .h5 to .tflite.
`toco --keras_model_file yolov2tiny.h5 --output_format TFLITE --inference_type FLOAT --output_file model.tflite` 
The file size does not change much before and after this conversion(60MB).
If inference_type is not FLOAT, the size can be reduced, but an error occurs in nncase.

> Fatal: expect FLOAT32 tensor but got INT8, use '--inference_type=FLOAT' when converting via toco.

Try converting nncase.
Images taken with M5StickV are stored in the "images" directory.
`./ncc -i tflite -o k210model --dataset images model.tflite model.kmodel`

> 2020-01-17 01:45:47.659128: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU 

> supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX 

> AVX2 FMA

> 0: InputLayer -> 1x3x240x320

> 1: K210Conv2d 1x3x240x320 -> 1x16x120x160

> 2: K210Conv2d 1x16x120x160 -> 1x32x60x80

> 3: K210Conv2d 1x32x60x80 -> 1x64x30x40

> 4: K210Conv2d 1x64x30x40 -> 1x128x15x20

> 5: K210Conv2d 1x128x15x20 -> 1x256x15x20

> 6: QuantizedMaxPool2d 1x256x15x20 -> 1x256x8x10

> 7: K210Upload 1x256x8x10 -> 1x256x8x10

> 8: K210Conv2d 1x256x8x10 -> 1x512x8x10

> 9: K210Conv2d 1x512x8x10 -> 1x1024x8x10

> 10: K210Conv2d 1x1024x8x10 -> 1x1024x8x10

> 11: K210Conv2d 1x1024x8x10 -> 1x50x8x10

> 12: Dequantize 1x50x8x10 -> 1x50x8x10

> 13: OutputLayer 1x50x8x10

> KPU memory usage: 2097152 B

> Main memory usage: 97280 B

The conversion was successful, but the model size is 15GB.
Size has been reduced by 1/4.
Still, it is too big to read on M5Stick.

Please tell me the effective method.

Thanks.

**Comment by sunnycase** [comment]
You can replace the backbone of your model with mobilenet v1 or v2.

**Comment by eraser3112** [comment]
Thank you for answering.

In my view, mobilenet only supports classification.
Object detection can be done with yolo.
Isn't object detection possible?

Please tell me if you are wrong.

**Comment by sunnycase** [comment]
classification is softmax not mobilenet, you can use mobilenet + yolo layer to detect objects.
________________________________

**Comment by eraser3112** [comment]
Thanks!
I do not know the specific method.
Still, you told me that the network structure needed to be reviewed.
Close the issue.
